sdk/trace/span.go

```diff
@@ -496,6 +497,20 @@ func (s *recordingSpan) End(options ...trace.SpanEndOption) {
 	}
 	s.mu.Unlock()
 
+	defer func() {
+		if s.tracer.selfObservabilityEnabled {
+			var samplingResultAttr attribute.KeyValue
+			if s.spanContext.IsSampled() {
+				samplingResultAttr = otelconv.SDKSpanEnded{}.AttrSpanSamplingResult(
```

Comment:
You should be able to use `s.tracer.spanLiveMetric.AttreSpanSamplingResult`, right?  That also ensures the sampling result attribute is supposed to be present on the live span metric.

Reply:
Addressed 6304e1d9